server/commands/teamCreator.ts

```diff
@@ -60,19 +59,23 @@ async function teamCreator({
     // If the self-hosted installation has a single team and the domain for the
     // new team is allowed then assign the authentication provider to the
     // existing team
-    if (teamCount === 1 && domain && isDomainAllowed(domain)) {
+    if (teamCount === 1 && domain) {
       const team = await Team.findOne();
       invariant(team, "Team should exist");
 
-      authP = await team.$create<AuthenticationProvider>(
-        "authenticationProvider",
-        authenticationProvider
-      );
-      return {
-        authenticationProvider: authP,
-        team,
-        isNewTeam: false,
-      };
+      if (await team.isDomainAllowed(domain)) {
+        authP = await team.$create<AuthenticationProvider>(
+          "authenticationProvider",
+          authenticationProvider
+        );
+        return {
+          authenticationProvider: authP,
+          team,
+          isNewTeam: false,
+        };
+      } else {
+        throw AuthenticationError(`Domain ${domain} is not on the allow list`);
```

Comment:
This was replicating the old code path as closely as possible, except I added in this slightly more descriptive error message

Reply:
This should probably be the new `DomainNotAllowedError` though, ye?